Cargo.toml

```diff
@@ -13,6 +13,8 @@ license = "MIT/Apache-2.0"
 [dependencies]
 chrono = { version = "0.4", default-features = false }
 serde = { version = "1", optional = true, default-features = false }
+phf = { version = "0.10", default-features = false, features = ["uncased"] }
+uncased = { version = "0.9", default-features = false }
```

Comment:
hm, if this goes in as a feature flag and a crate doesn't enable it, you can still accidentally get case insensitive matching if any other crate in your dependencies uses `chrono-tz` with that feature enabled.

for my curiosity, why is a minor bump something to avoid?

Reply:
Hmm I was thinking of the minor bump as being visible to external users, but that's not the case here, so yeah that is fine.

I do think that this should be behind _a_ feature flag. It adds a new dependency and some possibly surprising behavior, it also allows us to merge this without caring about the performance implications.

I agree that if folks have a dependency with the feature enabled then they have a choice of either removing the dep or dealing with uncased strings. That's still more of a choice than just forcing them to use uncased strings, though.

Reply:
I still think `Tz`'s `FromStr` implementation should have a single meaning regardless of feature flags just because it's hard to reason about what features get enabled from any dependency in the dependency tree.

How about this alternative proposal:

1. Keep the case-sensitive behaviour of `FromStr` as is, but use perfect hash tables for its implementation. This change can go directly in the crate without a feature flag since it doesn't change the API semantics at all, it's just a performance optimization
2. Make a separate `Tz::from_str_uncased(&str)` method that is behind a `uncased_09` feature flag for the users that want to parse uncased timezones.

This way users can choose exactly the API they want and cannot step on each other's feet

Reply:
Oh, that does sound better, that would be great.

Reply:
@quodlibetor I just pushed an implementation of the above, let me know what you think :)